engine-kommand/src/commonMain/kotlin/com/saveourtool/processbuilder/KommandEngine.kt

```diff
@@ -0,0 +1,66 @@
+package com.saveourtool.processbuilder
+
+import com.saveourtool.processbuilder.utils.CurrentOs
+import com.saveourtool.processbuilder.utils.cwd
+import com.saveourtool.processbuilder.utils.getCurrentOs
+import com.saveourtool.processbuilder.utils.logger
+import com.saveourtool.processbuilder.utils.stderr
+import com.saveourtool.processbuilder.utils.stdout
+
+import com.kgit2.process.Command
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+
+private typealias CommandWithArgs = Pair<String, Array<String>>
+
+object KommandEngine : Engine {
+    private val scope = CoroutineScope(Dispatchers.Default)
+
+    override suspend fun execute(
+        command: String,
+        config: ProcessBuilderConfig,
+    ): ExecutionResult {
+        val (cmd, args) = getCommandWithArgs(escapeDoubleTicks(command)).also {
```

Comment:
I'd also decompose to `cmd` and `args` within the `also { ... }` block.

Reply:
will print it as string (command) and list of strings (args)